encrypt/ibe/ibe.go

```diff
@@ -132,6 +132,94 @@ func DecryptCCAonG1(s pairing.Suite, private kyber.Point, c *Ciphertext) ([]byte
 	return msg, nil
 }
 
+// EncryptCCAonG1WithAD encrypts with additional data bound via commitment.
+// different AD values produce different ciphertexts for replay attack prevention.
+func EncryptCCAonG1WithAD(s pairing.Suite, master kyber.Point, ID, msg, additionalData []byte) (*Ciphertext, error) {
```

Comment:
Why copy-paste the whole code instead of doing like for `h3` where you just add the extra parameter and use that new function with an empty new parameter for the old function?